### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by kratz00**
Just some small fixes for some compiler warnings I stumbled across.

### CODE REVIEW

File: heimdall-frontend/source/mainwindow.cpp [hunk 1]
```diff
@@ -659,7 +659,7 @@ void MainWindow::SelectPartitionName(int index)
 
 		partitionFileGroup->setTitle(title);
 
-		if (!fileInfo.GetFilename().isEmpty())
+		if (pitEntry && !fileInfo.GetFilename().isEmpty())
```
Comment (Benjamin-Dobell) [inline heimdall-frontend/source/mainwindow.cpp]: Hmm, pitEntry should not be null here. If it is then there's a problem elsewhere. Did you encounter this personally?

**Comment by Benjamin-Dobell** [comment]
The only change I'm unsure about is the `pitEntry` null check. Everything else looks good. Well typically I put a space in `delete[]` (i.e. `delete []`), but it's not that big of a deal. I'll merge the whole lot if you can explain the `pitEntry` change.

File: heimdall-frontend/source/mainwindow.cpp (hunk 1 above)
Comment (kratz00) [inline heimdall-frontend/source/mainwindow.cpp]: I did not encounter this personally.
Just by looking at the code, you also see a check the pitEntry is not null in line 657.
So if pitEntry can be null there, it also can be null in line 664.
At least that was my reasoning behind at the null check :)

Comment (Benjamin-Dobell) [inline heimdall-frontend/source/mainwindow.cpp]: Fair enough. I imagine the original check was just me being a lazy copy 'n' paste programmer - which was the cause of most the other bugs you've fixed too. Whoops!

**Pull request merged**